src/shared/components/msktable/MSKTable.tsx

```diff
@@ -0,0 +1,126 @@
+import SimpleTable from "../simpleTable/SimpleTable";
+import * as React from 'react';
+import {observable, computed, action} from "mobx";
+import {observer} from "mobx-react";
+import './styles.scss';
+
+export type Column<T> = {
+    name: string;
+    filter?:(data:T, filterString:string)=>boolean;
+    sort?:(a:T, b:T)=>number;
+    render:(data:T)=>JSX.Element;
+};
+
+type MSKTableProps<T> = {
+    columns:Column<T>[];
+    data:T[];
+};
+
+class MSKTableStore<T> {
+    @observable public filterString:string;
+    @observable public page:number;
+    @observable public itemsPerPage:number;
+    @observable public sortColumn:string;
+    @observable public sortAscending:boolean;
+    @observable public columns:Column<T>[];
+    @observable.ref public data:T[];
+    @computed get sortedFilteredData():T[] {
```

Comment:
after a chat in person, onur agreed this seems at the moment to be a good solution, flexible enough to handle things